sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/AvailabilityStrategy.java

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.List;
+
+/**
+ * The type Availability strategy.
+ */
+public class AvailabilityStrategy {
```

Comment:
should this be interface class?

Reply:
I would prefer interface

Reply:
If we got with interface,  won't it allow users to implement it right away and pass it in the API? How would we handle it since there are no APIs in this interface. Seems to be just a marker interface to me as of now.

Reply:
Made it abstract class. Having a package private constructor doesnt let users subclass it